app/controllers/DeclarationController.scala

```diff
@@ -96,8 +89,10 @@ class DeclarationController @Inject() (
       val taxYearStartDate = Some(reportDetails.periodStart.toString)
       val redirect = Redirect(navigator.nextPage(DeclarationPage(srn), NormalMode, request.userAnswers))
 
-      val journeyType =
-        JourneyType.Standard // TODO: pass in JourneyType based on actual journey, currently submission is only for standard journey
+      val journeyType = request.userAnswers.get(ViewChangeQuestionPage(srn)) match {
+        case Some(TypeOfViewChangeQuestion.ChangeReturn) => JourneyType.Amend
+        case _ => JourneyType.Standard
```

Comment:
is the other option `ViewReturn` and is this not still the amend journey?

Reply:
ViewReturn Route had no declaration. Only view and Download.

Reply:
Yes, this question is not asked or answered as part of the standard journey

Reply:
Correct, I am not saying it will not work. It is confusing and unclear as this question has nothing to do with the standard journey, rather is a hack to get this to work without passing some other variable ;)

Reply:
We have impure code in many places - just another I guess. The main drawback I see here is with navigation. It might be possible to navigate to the declaration page without this question being answered (in the frontend cache) and submit with the wrong journey type...